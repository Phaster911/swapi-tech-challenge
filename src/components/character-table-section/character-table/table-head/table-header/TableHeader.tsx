import styled from "styled-components";
import { COLORS, controlBorderRadius } from "../../../../../style-constants";

const TableHeader = styled.th`
  --radius: calc(${controlBorderRadius} * 2);
  background-color: ${COLORS.mainBackground};
  text-align: left;
  padding: 1.5rem 0.5rem;

  &:first-of-type {
    border-top-left-radius: var(--radius);
    border-bottom-left-radius: var(--radius);
  }
  &:last-of-type {
    border-top-right-radius: var(--radius);
    border-bottom-right-radius: var(--radius);
  }
`;

export default TableHeader;
