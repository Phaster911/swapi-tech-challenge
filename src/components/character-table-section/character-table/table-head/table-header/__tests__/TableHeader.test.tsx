import { render } from "@testing-library/react";
import TableHeader from "../TableHeader";

describe("<TableHeader />", () => {
  it("renders the component", () => {
    const { container } = render(
      <table>
        <thead>
          <tr>
            <TableHeader>ABC</TableHeader>
          </tr>
        </thead>
      </table>
    );
    expect(container).toMatchSnapshot();
  });
});
