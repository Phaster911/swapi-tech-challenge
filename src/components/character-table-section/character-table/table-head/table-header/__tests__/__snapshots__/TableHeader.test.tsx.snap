// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TableHeader /> > renders the component 1`] = `
.c0 {
  --radius: calc(5px * 2);
  background-color: #e9ecee;
  text-align: left;
  padding: 1.5rem 0.5rem;
}

.c0:first-of-type {
  border-top-left-radius: var(--radius);
  border-bottom-left-radius: var(--radius);
}

.c0:last-of-type {
  border-top-right-radius: var(--radius);
  border-bottom-right-radius: var(--radius);
}

<div>
  <table>
    <thead>
      <tr>
        <th
          class="c0"
        >
          ABC
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
